eslint-bridge/tests/services/analysis/analyzers/yaml/analyzer.test.ts

```diff
@@ -186,4 +188,37 @@ describe('analyzeYAML', () => {
     });
     expect(issues).toHaveLength(0);
   });
+
+  it('should provide the filename composed of itself and the AWS lambda function to the rule context', async () => {
+    const seenFunctionNames = [];
+    const filenameWatchRule: Rule.RuleModule = {
+      create(context: Rule.RuleContext) {
+        return {
+          Program: _node => {
+            const composedFilename = context.getFilename();
+            seenFunctionNames.push(composedFilename);
+          },
+        };
+      },
+    };
+    initializeLinter([
+      { key: 'filename-watch-rule', configurations: [], fileTypeTarget: ['MAIN'] },
+    ]);
+    linter.linter.defineRule('filename-watch-rule', filenameWatchRule);
+    const filePath = join(fixturesPath, 'functionNames.yaml');
+    const functionNames = ['SomeLambdaFunction', 'SomeServerlessFunction'];
+    const filenames = composeAwsFunctionFilenames(filePath, functionNames);
+    await analyzeYAML({
+      filePath,
+      fileContent: undefined,
+    });
+
+    const [firstSeenFunction, secondSeenFunction] = seenFunctionNames;
+    expect(firstSeenFunction).toEqual(filenames[0]);
+    expect(secondSeenFunction).toEqual(filenames[1]);
+
+    function composeAwsFunctionFilenames(filePath, functionNames) {
+      return functionNames.map(composeAwsFunctionFilename.bind(null, filePath));
+    }
+  });
```

Comment:
Let's keep the test simple by testing a single resource since it is already validated for two resources by another layer test. That would avoid us having to collect filenames during the rule execution and asserting their value afterwards. 

```suggestion
  it('should provide a synthetic filename to the rule context', async () => {
    expect.assertions(1);

    const resource = 'SomeLambdaFunction';
    const filePath = join(fixturesPath, 'synthetic-filename.yaml');
    const syntheticFilename = composeSourceCodeFilename(filePath, resource);

    const rule = {
      key: 'synthetic-filename',
      module: {
        create(context: Rule.RuleContext) {
          return {
            Program: () => {
              const filename = context.getFilename();
              expect(filename).toEqual(syntheticFilename);
            },
          };
        },
      }
    }

    initializeLinter([{ key: rule.key, configurations: [], fileTypeTarget: ['MAIN'] }]);
    linter.linter.defineRule(rule.key, rule.module);

    await analyzeYAML({ filePath, fileContent: undefined});
  });
```